api/geoLoadingSlice.ts

```diff
@@ -0,0 +1,21 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+export interface geoLoadingState {
+  isLoading: boolean
+}
+
+const initialState: geoLoadingState = {
+  isLoading: false,
+};
+
+const geoLoadingSlice = createSlice({
+  name: "geoLoading",
+  initialState,
+  reducers: {
+    getIsGeoLoading: (state, action: PayloadAction<boolean>) => {
+      state.isLoading = action.payload;
+    },
```

Comment:
We might want to change the language on this reducer name, as it's called `getIsGeoLoading`, but what it does is set a value rather than get a value.  We should rename this to `setIsGeoLoading`.